brainstormforce/ultimate-addons-for-gutenberg
# Improvement: Added Forms recaptcha settings on dashboard and used globally

### Description

We have introduced a new setting in the admin dashboard where users can add the google Recaptcha site and secret key.
Which will be accessible throughout the site where forms blocks are added. 

- Added forms Recaptcha site and secret key option on the dashboard
  - Created separate files and functions
- Updated front-end logic for forms Recaptcha.
- Updated render and save.js file

### Types of changes
breaking change which adds functionality

### Test case 1

- Switch to master branch 
- drag and drop forms block on different pages
- Apply Recaptcha settings to every forms block (Version, site key, and secret key )
- Switch to the forms-site-secret-key branch
- Check site and secret key value are automatically filled or not
- Check forms block with different Recaptcha versions are working or not on the front-end.

### Test Case 2 

- For new users, Switch to forms-site-secret-key branch
- Drag and drop forms block
- Enable ReCaptcha setting from the setting panel
- Add site and secret key in dashboard setting as per the versions
- check on front-end where forms working properly or not

### Recaptcha Test Cases

- Check all the settings which we have added for Recaptcha like hide Recaptcha
- On front-end check modifying id = g-recaptcha-response response value by inspecting and then submitting the form
- Delete the g-recaptcha-response element by inspecting and try submitting the form.
- Enter the wrong site and secret key on the dashboard and check form is working properly or not

### Checklist:
